src/repository/apmRepository.ts

```diff
@@ -61,6 +61,22 @@ export class ApmRepository {
     });
   }
 
+   /**
+   * Get the registry contract for an ENS domain and the registry ABI.
+   * It will slice the first subdomain and query the rest as the registry domain.
+    * ENS domain:      admin.dnp.dappnode.eth
+    * Registry domain:       dnp.dappnode.eth
+   * @param ensName - The ENS domain name, e.g., "admin.dnp.dappnode.eth".
+   * @param registryAbi - The ABI of the registry contract in JSON format.
+   * @returns A contract instance of the registry for the specified ENS domain, or null if the registry address is not resolved.
+   */
+  public async getRegistryContract(ensName: string, registryAbi: string): Promise<ethers.Contract | null> {
```

Comment:
The ideal solution goues through adapting the consumers to use the Repo contract provided by the toolkit, and in last instance typechain. Does not make sense at all to have a function that return an instance of the library ethers, we are not providing any additional value here, the consumer would rather prefer to create the instance directly from the ethers library